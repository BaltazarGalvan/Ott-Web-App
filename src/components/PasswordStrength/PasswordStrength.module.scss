@use '../../styles/variables';
@use '../../styles/theme';

.passwordStrength {
  position: relative;
  margin: variables.$base-spacing 0;
  font-family: theme.$body-font-family;
  font-size: 14px;
}

.passwordStrengthBar {
  position: relative;
  width: 170px;
  height: 6px;
  margin: variables.$base-spacing 0;

  background: #ddd;
  border-radius: 5px;
}

.passwordStrengthFill {
  position: absolute;
  width: 0;
  height: inherit;
  background: transparent;
  border-radius: inherit;
  transition: width 0.5s ease-in-out, background 0.25s;
}

.passwordStrengthFill[data-strength='1'] {
  width: 25%;
  background: orangered;
}

.passwordStrengthFill[data-strength='2'] {
  width: 50%;
  background: orange;
}

.passwordStrengthFill[data-strength='3'] {
  width: 75%;
  background: yellowgreen;
}

.passwordStrengthFill[data-strength='4'] {
  width: 100%;
  background: green;
}
