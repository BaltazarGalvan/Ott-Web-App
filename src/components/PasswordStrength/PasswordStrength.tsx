import React from 'react';
import { useTranslation } from 'react-i18next';

import styles from './PasswordStrength.module.scss';

type Props = {
  password: string;
};

const PasswordStrength: React.FC<Props> = ({ password }: Props) => {
  const { t } = useTranslation('account');
  const passwordStrength = (password: string) => {
    let strength = 0;

    if (password.match(/[a-z]+/)) {
      strength += 1;
    }
    if (password.match(/[A-Z]+/)) {
      strength += 1;
    }
    if (password.match(/[0-9|!@#$%^&*()_+-=]+/)) {
      strength += 1;
    }
    if (password.length >= 6) {
      strength += 1;
    }

    return strength;
  };
  return (
    <div className={styles.passwordStrength}>
      <div className={styles.passwordStrengthBar}>
        <div className={styles.passwordStrengthFill} data-strength={passwordStrength(password)}></div>
      </div>
      <span>{t('registration.password_strength')}</span>
    </div>
  );
};

export default PasswordStrength;
