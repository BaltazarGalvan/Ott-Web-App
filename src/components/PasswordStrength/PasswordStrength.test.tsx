import React from 'react';
import { render } from '@testing-library/react';

import PasswordStrength from './PasswordStrength';

describe('<PasswordStrength>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<PasswordStrength />);

    expect(container).toMatchSnapshot();
  });
});
