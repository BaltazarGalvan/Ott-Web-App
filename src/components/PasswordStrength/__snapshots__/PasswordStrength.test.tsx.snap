// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordStrength> renders and matches snapshot 1`] = `
<div>
  <div
    class="passwordStrength"
    data-strength="2"
  >
    <div
      class="passwordStrengthBar"
    >
      <div
        class="passwordStrengthFill"
      />
    </div>
     
    <span
      class="label"
    >
      registration.password_strength.fair
    </span>
  </div>
</div>
`;
